### PR TITLE
Fixed the hotkeys so that they work with Render to Main

### PR DESCRIPTION
**Pull request by skidau**

### CODE REVIEW

File: Source/Core/DolphinWX/Frame.cpp [hunk 1]
```diff
@@ -1420,25 +1420,22 @@ void CFrame::ParseHotkeys(wxKeyEvent &event)
...
+			if (IsHotkey(event, i))
 			{
-				if (IsHotkey(event, i))
+				int cmd = GetCmdForHotkey(i);
+				if (cmd >= 0)
 				{
-					int cmd = GetCmdForHotkey(i);
-					if (cmd >= 0)
+					wxCommandEvent evt(wxEVT_MENU, cmd);
+					wxMenuItem *item = GetMenuBar()->FindItem(cmd);
```
Comment (Buddybenj) [inline Source/Core/DolphinWX/Frame.cpp]: Asterisk against type

**Comment by RachelBryk** [comment]
>Reduced the hotkey checking rate from 60 times a sec to 10 times a sec

Change this back, the delay is unacceptable when using hotkeys rapidly.

File: Source/Core/Core/HotkeyManager.cpp [hunk 2]
```diff
@@ -266,9 +258,12 @@ std::string HotkeyManager::GetName() const
 
 void HotkeyManager::GetInput(HotkeyStatus* const kb)
 {
-	for (int set = 0; set < 6; set++)
+	for (int set = 0; set < 3; set++)
```
Comment (BhaaLseN) [inline Source/Core/Core/HotkeyManager.cpp]: Not in the style guide, but I'd personally prefer curly braces on the outer block (in this case, the ```for``` loop) too if the inner block needs them for whatever reason. Any comments on that (@Lioncash perhaps)?

File: Source/Core/Core/HotkeyManager.h [hunk 3]
```diff
@@ -34,6 +34,7 @@ namespace HotkeyManagerEmu
 	void Shutdown();
 
 	InputConfig* GetConfig();
-	void GetStatus(u8 _port, HotkeyStatus* _pKeyboardStatus);
+	void GetStatus();
+	bool OwnsLock();
```
Comment (BhaaLseN) [inline Source/Core/Core/HotkeyManager.h]: Mostly a matter of taste, but does the outside need to know about the locking going on? Maybe rename it to ```IsReady```/```IsBusy```` or something like that instead? Fine to ignore tho.

**Comment by RachelBryk** [comment]
> Removed the requirement to have the Freelook option enabled for the freelook hotkeys to work.

This is annoying because you cannot disable it now without unmapping the hotkeys, then remapping them when you want to use it...

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

**Comment by cegli** [comment]
@RachelBryk 
I like the change because it's consistent.  None of the other hotkeys have a separate checkbox hidden away somewhere that enables or disables them.  Users may also get confused, because they would assign the keys, but they wouldn't work because the freelook check mark isn't checked.  I would think the typical use case would be to bind them to keys you don't typically use, and not use them if they the feature is not needed.

Is there a time that having the keys assigned causes issues?  I'm very interested because we use freelook extensively in Dolphin VR and want to make sure I'm not missing something.

**Comment by magumagu** [comment]
@cegli The commit in question doesn't remove the freelook checkbox; it just lets the user modify freelook projection matrices while freelook isn't enabled. (This does nothing visible, but has an effect if the user turns on freelook later).

**Comment by cegli** [comment]
@magumagu Right, the freelook checkbox needs to remain for the mouse free look, but now it's no longer needed for keyboard or whatever is assigned to the freelook hotkeys.  We've done the same thing in Dolphin VR, and renamed that checkbox to "Mouse Freelook" to make it clear it doesn't affect the hotkeys.

That's something that should probably be done in this pull by the way.  If the freelook checkbox is no longer needed for the freelook hotkeys, it should be renamed something like "Mouse Freelook" and the description should be updated.  I think it still references static keyboard keys at this point.

**Comment by skidau** [comment]
The bFreelook logic isn't clear to me.  I am going to leave the check in VertexShaderManager for now.

**Comment by RachelBryk** [comment]
@cegli The other hotkeys don't cause annoying and unexpected effects if pressed accidentally. They are also mapped by default. And the default mapping conflicts with the default controller mappings. Also, background input.

**Comment by skidau** [comment]
The freelook hotkeys are no longer mapped by default.  None of the hotkeys are mapped by default anymore as they were causing confusion with double presses.

**Comment by cegli** [comment]
@RachelBryk Good points.  The default mapping should probably either be changed or removed if it is changed so they don't need to be explicitly enabled.

@skidau Fair enough!  I'm fairly certain it checks in three places:

1. Once in CFrame::OnMouse(), where it should be left alone, considering we have no other way to pan/adjust tilt
2. Once in CFrame::PollHotkeys() where it checks updates the s_fViewTranslationVector, but doesn't actually apply any freelook.
3. It gets checked per frame in the VertexShaderManager::SetConstatnts() where it actually applies the freelook translationvector (Matrix44::Translate(mtxA, s_fViewTranslationVector).

So I guess the way it is right now lets you update the s_fViewTranslationVector, but doesn't actually apply it.

**Pull request merged**